src/vs/workbench/contrib/codeEditor/browser/saveParticipants.ts

```diff
@@ -273,7 +275,50 @@ class CodeActionOnSaveParticipant implements ITextFileSaveParticipant {
 		@IConfigurationService private readonly configurationService: IConfigurationService,
 		@IInstantiationService private readonly instantiationService: IInstantiationService,
 		@ILanguageFeaturesService private readonly languageFeaturesService: ILanguageFeaturesService,
-	) { }
+		@IHostService private readonly hostService: IHostService,
+		@IEditorService private readonly editorService: IEditorService,
+		@ICodeEditorService private readonly codeEditorService: ICodeEditorService,
+	) {
+		this.hostService.onDidChangeFocus(async () => {
+			this.triggerCodeActionsCommand();
+		});
+
+		this.editorService.onDidActiveEditorChange(async () => {
+			this.triggerCodeActionsCommand();
+		});
```

Comment:
The async here doesn't do anything

Reply:
yep left it in from when i wasn't using the helper func 👍